soc: nordic: nrf54l: fix configuration of DCDC regulator

nika-nordic opened the pull request:
DCDC regulator on nRF54L may not always works as intended. Apply a fix addressing that.

nika-nordic committed:
soc: nordic: nrf54l: fix configuration of DCDC regulator

DCDC regulator on nRF54L may not always works as intended.
Apply a fix addressing that.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

gmarull commented on soc/nordic/nrf54l/soc.c:
```diff
@@ -153,6 +153,11 @@ static int nordicsemi_nrf54l_init(void)
 	}
 
 #if (DT_PROP(DT_NODELABEL(vregmain), regulator_initial_mode) == NRF5X_REG_MODE_DCDC)
+#if defined(__CORTEX_M) && !defined(NRF_TRUSTZONE_NONSECURE) && defined(__ARM_FEATURE_CMSE)
```
question: is vregmain available from riscv core? if not the cortex_m bit may be redundant.

nika-nordic committed:
soc: nordic: nrf54l: tune configuration of DCDC regulator

DCDC regulator on nRF54L may not always works as intended.
Tune the fix addressing that.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

anangl commented on soc/nordic/nrf54l/soc.c:
```diff
@@ -153,6 +153,11 @@ static int nordicsemi_nrf54l_init(void)
 
 #if (DT_PROP(DT_NODELABEL(vregmain), regulator_initial_mode) == NRF5X_REG_MODE_DCDC)
+#if defined(__CORTEX_M) && !defined(NRF_TRUSTZONE_NONSECURE) && defined(__ARM_FEATURE_CMSE)
+	if (*(uint32_t *)0x00FFC334 <= 0x180A1D00) {
+		*(uint32_t *)0x50120640 = 0x1EA9E040;
```
Don't we need `uint32_t volatile *` here?

nika-nordic replied:
you are right, original fix was missing that
